plugins/modules/web_app_pool.ps1

```diff
@@ -0,0 +1,347 @@
+        $attribute_key_split = $attribute_key_split[1..$($attribute_key_split.Length - 1)]
+        $parent = $pool.$attribute_parent
+
+        foreach ($key in $attribute_key_split) {
+            $attribute_meta = $parent.Attributes | Where-Object { $_.Name -eq $key }
+            $parent = $parent.$key
+            if ($null -eq $attribute_meta) {
+                $attribute_meta = $parent
+            }
+        }
+        $attribute_child = $attribute_key_split[-1]
+    }
+
+    if ($attribute_meta) {
+        if (($attribute_meta.PSObject.Properties.Name -eq "Collection").Count -gt 0) {
+            return , (Convert-CollectionToList -collection $attribute_value)
+        }
+        $type = $attribute_meta.Schema.Type
+        $value = $attribute_value
+        if ($type -eq "enum") {
+            # Attempt to convert the value from human friendly to enum value - use existing value if we fail
+            $dot_net_class = Get-DotNetClassForAttribute -attribute_parent $attribute_parent
+            $enum_attribute_name = $attribute_child.Substring(0, 1).ToUpper() + $attribute_child.Substring(1)
+            $enum = $dot_net_class.GetProperty($enum_attribute_name).PropertyType.FullName
+            if ($enum) {
+                $enum_values = [Enum]::GetValues($enum)
+                foreach ($enum_value in $enum_values) {
+                    if ($attribute_value.GetType() -is $enum_value.GetType()) {
+                        if ($enum_value -eq $attribute_value) {
+                            $value = $enum_value
+                            break
+                        }
+                    }
+                    else {
+                        if ([System.String]$enum_value -eq [System.String]$attribute_value) {
+                            $value = $enum_value
+                            break
+                        }
+                    }
+                }
+            }
+        }
+        # Try and cast the variable using the chosen type, revert to the default if it fails
+        Set-Variable -Name casted_value -Value ($value -as ([type] $attribute_meta.TypeName))
+        if ($null -eq $casted_value) {
+            $value
+        }
+        else {
+            $casted_value
+        }
+    }
+    else {
+        $attribute_value
+    }
+}
+
+# Ensure WebAdministration module is loaded
+if ($null -eq (Get-Module -Name "WebAdministration" -ErrorAction SilentlyContinue)) {
+    Import-Module WebAdministration
+    $web_admin_dll_path = Join-Path $env:SystemRoot system32\inetsrv\Microsoft.Web.Administration.dll
+    Add-Type -LiteralPath $web_admin_dll_path
+}
+
+$pool = Get-Item -LiteralPath IIS:\AppPools\$name -ErrorAction SilentlyContinue
+if ($state -eq "absent") {
+    # Remove pool if present
+    if ($pool) {
+        try {
+            Remove-WebAppPool -Name $name -WhatIf:$check_mode
+        }
+        catch {
+            $module.FailJson("Failed to remove Web App pool $($name): $($_.Exception.Message)")
```

Comment:
A nice feature of `FailJson` is it accepts a second argument which we can use for providing better error messages. We just need to add `$_` as an argument and `FailJson` will do the rest for us.

```suggestion
            $module.FailJson("Failed to remove Web App pool $($name): $($_.Exception.Message)", $_)
```

An alternative is to just remove the `try/catch` altogether but sometimes it is nice to have a more descriptive error message.